### PR TITLE
IRIS-5080 | Implement automated tests for confirming email address for registered user

### PR DESCRIPTION
**Pull request by tortila**
[Ticket](https://wikia-inc.atlassian.net/browse/IRIS-5080)
- new test group + suite
- new jenkins job
- new email account dedicated for this test

### CODE REVIEW

File: src/test/java/com/wikia/webdriver/common/core/EmailUtils.java [hunk 1]
```diff
@@ -22,12 +22,14 @@
...
+  private static final String CONFIRM_EMAIL_LINK = String.format(EMAIL_LINK_PATTERN, "Confirm Now");
 
   // Pattern.DOTALL flag forces dot in regex to also match line terminators
-  private static Pattern PASSWORD_RESET_PATTERN =
-    Pattern.compile(PASSWORD_RESET_LINK, Pattern.DOTALL);
+  private static Pattern getPatternForString(String value) {
+    return Pattern.compile(value, Pattern.DOTALL);
+  }
 
   private EmailUtils() {
```
Comment (WikiaTech) [inline src/test/java/com/wikia/webdriver/common/core/EmailUtils.java]: MINOR Move this constructor to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 6 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L157](https://github.com/Wikia/selenium-tests/blob/8be75c6c6729e2997fc25803f6480d64d75263f2/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L157): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L173](https://github.com/Wikia/selenium-tests/blob/8be75c6c6729e2997fc25803f6480d64d75263f2/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L173): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L594](https://github.com/Wikia/selenium-tests/blob/8be75c6c6729e2997fc25803f6480d64d75263f2/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L594): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L648](https://github.com/Wikia/selenium-tests/blob/8be75c6c6729e2997fc25803f6480d64d75263f2/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L648): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)
1. ![MINOR][MINOR] [EmailUtils.java#L43](https://github.com/Wikia/selenium-tests/blob/8be75c6c6729e2997fc25803f6480d64d75263f2/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L43): Define a constant instead of duplicating this literal "imaps" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [EmailUtils.java#L62](https://github.com/Wikia/selenium-tests/blob/8be75c6c6729e2997fc25803f6480d64d75263f2/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L62): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)
1. ![MINOR][MINOR] [EmailUtils.java#L108](https://github.com/Wikia/selenium-tests/blob/8be75c6c6729e2997fc25803f6480d64d75263f2/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L108): Move the array designator from the variable to the type. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1197)
1. ![MINOR][MINOR] [EmailUtils.java#L131](https://github.com/Wikia/selenium-tests/blob/8be75c6c6729e2997fc25803f6480d64d75263f2/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L131): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. ![MINOR][MINOR] [UserWithEmailFactory.java#L13](https://github.com/Wikia/selenium-tests/blob/8be75c6c6729e2997fc25803f6480d64d75263f2/src/test/java/com/wikia/webdriver/common/core/helpers/UserWithEmailFactory.java#L13): Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)
1. ![INFO][INFO] [WikiBasePageObject.java#L161](https://github.com/Wikia/selenium-tests/blob/8be75c6c6729e2997fc25803f6480d64d75263f2/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L161): Complete the task associated to this TODO comment. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/test/java/com/wikia/webdriver/common/core/EmailUtils.java [hunk 2]
```diff
@@ -86,6 +88,7 @@ public static String getFirstEmailContent(String userName, String password, Stri
       throw new WebDriverException();
     } catch (MessagingException | IOException | InterruptedException e) {
       PageObjectLogging.log("getFirstEmailContent", e, false);
+      PageObjectLogging.log("Cannot log in", String.format("cannot log in using credentials: %s/%s", userName, password), false);
```
Comment (rogatty) [inline src/test/java/com/wikia/webdriver/common/core/EmailUtils.java]: I'm not sure if we should show password in the logs. @ludwikkazmierczak?

**Comment by WikiaTech** [comment]
SonarQube analysis reported 11 issues
* ![MAJOR][MAJOR] 4 major
* ![MINOR][MINOR] 6 minor
* ![INFO][INFO] 1 info

Watch the comments in this conversation to review them.

#### 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BasePageObject.java#L157](https://github.com/Wikia/selenium-tests/blob/cd7eeafe452a26ec678fd23c7743a1ae67a8002f/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L157): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L173](https://github.com/Wikia/selenium-tests/blob/cd7eeafe452a26ec678fd23c7743a1ae67a8002f/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L173): Either log or rethrow this exception. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)
1. ![MAJOR][MAJOR] [BasePageObject.java#L594](https://github.com/Wikia/selenium-tests/blob/cd7eeafe452a26ec678fd23c7743a1ae67a8002f/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L594): Remove this unused private "getTabsCount" method. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AUnusedPrivateMethod)
1. ![MAJOR][MAJOR] [BasePageObject.java#L648](https://github.com/Wikia/selenium-tests/blob/cd7eeafe452a26ec678fd23c7743a1ae67a8002f/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/BasePageObject.java#L648): Remove this unused "initialTabsNumber" local variable. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1481)
1. ![MINOR][MINOR] [EmailUtils.java#L43](https://github.com/Wikia/selenium-tests/blob/cd7eeafe452a26ec678fd23c7743a1ae67a8002f/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L43): Define a constant instead of duplicating this literal "imaps" 4 times. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1192)
1. ![MINOR][MINOR] [EmailUtils.java#L62](https://github.com/Wikia/selenium-tests/blob/cd7eeafe452a26ec678fd23c7743a1ae67a8002f/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L62): Refactor this code to not nest more than 3 if/for/while/switch/try statements. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS134)
1. ![MINOR][MINOR] [EmailUtils.java#L107](https://github.com/Wikia/selenium-tests/blob/cd7eeafe452a26ec678fd23c7743a1ae67a8002f/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L107): Move the array designator from the variable to the type. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1197)
1. ![MINOR][MINOR] [EmailUtils.java#L130](https://github.com/Wikia/selenium-tests/blob/cd7eeafe452a26ec678fd23c7743a1ae67a8002f/src/test/java/com/wikia/webdriver/common/core/EmailUtils.java#L130): Move this trailing comment on the previous empty line. [rule](http://sonar-s4/coding_rules#rule_key=squid%3ATrailingCommentCheck)
1. ![MINOR][MINOR] [UserWithEmailFactory.java#L13](https://github.com/Wikia/selenium-tests/blob/cd7eeafe452a26ec678fd23c7743a1ae67a8002f/src/test/java/com/wikia/webdriver/common/core/helpers/UserWithEmailFactory.java#L13): Move this variable to comply with Java Code Conventions. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)
1. ![INFO][INFO] [WikiBasePageObject.java#L161](https://github.com/Wikia/selenium-tests/blob/cd7eeafe452a26ec678fd23c7743a1ae67a8002f/src/test/java/com/wikia/webdriver/pageobjectsfactory/pageobject/WikiBasePageObject.java#L161): Complete the task associated to this TODO comment. [rule](http://sonar-s4/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**